sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -665,7 +666,8 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
   def variableSubstituteDepth: Int = getConf(VARIABLE_SUBSTITUTE_DEPTH)
 
   def warehousePath: String = {
-    getConf(WAREHOUSE_PATH).replace("${system:user.dir}", System.getProperty("user.dir"))
+    getConf(WAREHOUSE_PATH).replace("${system:user.dir}",
+      new File(System.getProperty("user.dir")).toURI.toURL.toString).replace("//", "/")
```

Comment:
In my local mac osx, `new File(System.getProperty("user.dir")).toURI.toURL.toString)` will generate a file path with a trailling `/`. 

I am afraid this could be OS dependent? Thus, I did not change the default value.